the-incredible-bulk/src/main/java/gov/va/api/health/bulkfhir/service/controller/publication/OldestFirstFileToBuildManager.java

```diff
@@ -0,0 +1,68 @@
+package gov.va.api.health.bulkfhir.service.controller.publication;
+
+import gov.va.api.health.bulkfhir.service.filebuilder.FileBuildRequest;
+import gov.va.api.health.bulkfhir.service.filebuilder.FileBuilderExceptions.FindFileToBuildFailed;
+import gov.va.api.health.bulkfhir.service.filebuilder.FileToBuildManager;
+import gov.va.api.health.bulkfhir.service.status.StatusEntity;
+import gov.va.api.health.bulkfhir.service.status.StatusRepository;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@Builder
+@AllArgsConstructor(onConstructor = @__({@Autowired}))
+public class OldestFirstFileToBuildManager implements FileToBuildManager {
+
+  private final StatusRepository repository;
+
+  @Override
+  public FileBuildRequest getNextFileToBuild() {
+    try {
+      List<StatusEntity> availableFiles = repository.findByStatusNotStarted();
+      if (availableFiles == null || availableFiles.isEmpty()) {
+        /*
+         * No files that haven't been started yet have been found.
+         */
+        log.info("Didn't find any files available to be started.");
+        return null;
+      }
+
+      /*
+       * Sort the files
+       */
+      List<StatusEntity> sortedFiles = sortAvailableFiles(availableFiles);
+
+      StatusEntity fileToBuild = sortedFiles.get(0);
+
```

Comment:
... or, instead of copying stuff
```
Optional<StatusEntity> next = repository.findByStatusNotStarted().stream()
  .sort(compareByPublicationEpoch()) // <-- just return a comparator
  .findFirst();

if (next.isAbsent()) { 
  // no bueno ...
}

```